src/java.base/share/classes/com/sun/crypto/provider/KeyWrapCipher.java

```diff
@@ -230,13 +230,11 @@ protected void engineSetPadding(String padding)
     }
 
     /**
-     * Returns the block size (in bytes). i.e. 16 bytes.
-     *
-     * @return the block size (in bytes), i.e. 16 bytes.
+     * @return the block size (in bytes)
      */
     @Override
     protected int engineGetBlockSize() {
-        return cipher.getBlockSize();
+        return 8;
```

Comment:
Line 186 still says: 

`* symmetric cipher whose block size must be 128-bit`

Should that also be updated?

Reply:
Line 186 is correct in that the underlying Cipher block size must be 128-bit. However, the KW/KWP processing affected the input size requirement into 8-byte blocks, thus the overall cipher block size is now 8 instead of 16 bytes.

Reply:
So the block size is always 8 even when there is no padding?

Reply:
Yes, with KW (no pad), data must be in multiples of 8. When data does not meet this size requirement, an external padding scheme such as PKCS5/7 padding is needed to pad the data to multiples of 8 in order for KW mode to process the data.
As for KWP mode, it internally pads the data to multiples of 8 before starting the internal processing. Thus, no external padding is needed. Are you asking if 8 should be returned for KWP mode due to its internal padding? KWP is like a variant of KW, so it seems to me that it should return the same block size as KW.

Reply:
Your explanation make sense. I just wanted to make sure we didn't miss something.

Reply:
Yes, thanks for checking.